theme/models.py

```diff
@@ -324,4 +325,36 @@ def force_unique_emails(sender, instance, **kwargs):
         if sender.objects.filter(username=username).exclude(pk=instance.id).exists():
             raise ValidationError("Username already in use.")
 
+
 pre_save.connect(force_unique_emails, sender=User)
+
+
+@receiver(models.signals.pre_save, sender=UserProfile)
+def auto_delete_file_on_change(sender, instance, **kwargs):
+    """
+    Deletes old file from filesystem when corresponding UserProfile object is updated with new file.
+    """
+    if not instance.pk:
+        return
+
+    try:
+        up = UserProfile.objects.get(pk=instance.pk)
+        if up:
+            old_file_cv = up.cv
+            old_file_pic = up.picture
+            if not old_file_cv and not old_file_pic:
+                return
+        else:
+            return
+    except UserProfile.DoesNotExist:
+        return
+
+    new_file_cv = instance.cv
```

Comment:
Ok, so instance is the updated model and you're reading the old model from the db and comparing to see if it changed.  I think updating variable names and comments could help clarify what is going on.  If instance were renamed to updated_profile then I'd say removing assigning new_file_cv variable and just access it in the if statement below.